dagshub/auth/oauth.py

```diff
@@ -48,25 +40,96 @@ def oauth_flow(host: str, client_id: Optional[str] = None, referrer: Optional[st
     # Printing them line by line, because the link has to be its own print in order for Colab parser to correctly parse
     # the whole link
     print("\n\nOpen the following link in your browser to authorize the client:")
-    print(auth_link)
+    print(flow_data.auth_link)
     print("\n")
 
-    with rich_console.status("Waiting for authorization"):
-        res = httpx.post(f"{dagshub_url}/middleman", data={"request_id": middle_man_request_id}, timeout=None)
+    token = oauth_flow_post(flow_data)
+    logger.debug(f"Got token: {token}")
+    return token
+
 
+class OAuthFlowData(NamedTuple):
+    """Data object containing all necessary information for OAuth flow."""
+    auth_link: str
+    """
+    The link the user needs to visit to confirm the OAuth flow.
+    This link should be shown to the user, who needs to open it in their web browser.
```

Comment:
I'm usually using dataclasses everywhere and not named tuples, but I guess it still works so w/e